pydantic/dataclasses.py

```diff
@@ -230,6 +231,12 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         if frozen is not None:
             frozen_ = frozen
         else:
+            # It's not recommended to define both, as the setting from the dataclass decorator will take priority.
+            if config_wrapper.frozen:
+                warn(
+                    "Both 'dataclass' decorator and 'config.frozen' are defined. 'config.frozen' take priority.",
+                    UserWarning,
+                )
```

Comment:
This is not the case with your current implementation. In the `else` branch, `frozen` is guaranteed to be `None` (i.e. not defined), so they can't both be defined.

Imo we shouldn't raise any warning here, as long as we document the behavior.

Reply:
I'm inclined to raise a warning here - makes it more likely that folks can figure out the issue on their own + don't open an issue with a question.

Agreed with @Viicos though, this isn't logically in the right place given the warning.

Reply:
To be tested, but let's use `stacklevel=3` here, so that the warning is emitted at the right place (it should emit at the decorator usage)

Reply:
> Agreed with @Viicos though, this isn't logically in the right place given the warning.

I understand that the warning is displayed if config_wrapper.frozen is True, regardless of the value of frozen.
Is my understanding correct?

Reply:
We should only display the warning if `frozen` is set in the dataclass constructor, and it's set in `config`.

Reply:
Thank you! I fixed it!